src/main/java/com/safetynet/alerts/presentation/controller/FirestationController.java

```diff
@@ -0,0 +1,125 @@
+package com.safetynet.alerts.presentation.controller;
+
+import com.safetynet.alerts.logging.LogHandlerTiny;
+import com.safetynet.alerts.logic.service.FirestationService;
+import com.safetynet.alerts.presentation.model.SafetyAlertsModel;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+/**
+ * RestController for /firestation endpoint
+ *
+ * Includes POST/PUT/DELETE
+ */
+@RestController
+public class FirestationController extends BaseController{
+
+    @Autowired
+    private FirestationService firestationService;
+
+
+    /**
+     * Mapping for POST
+     *
+     * Returns:
+     * HttpStatus.CONFLICT if mapping already exists
+     * HttpStatus.INTERNAL_SERVER_ERROR if data file cannot be accessed
+     * Json string & HttpStatus.CREATED if successful
+     *
+     * @param address address for Firestation mapping
+     * @param station station number for Firestation mapping
+     * @return Json string & HttpStatus.CREATED if successful
+     */
+    @PostMapping("/firestation")
+    public ResponseEntity<String> addEntity(@RequestParam("address") String address, @RequestParam("station") int station){
+        //Log reqquest
+        logHandler.setLogger("FirestationController");
+        logHandler.logRequest("POST","/firestation", new String[] {address, String.valueOf(station)});
+
+        //confirm data is loaded
+        if (!safetyAlertsModel.isDataLoaded()){
```

Comment:
I would change it to `!firestationService.isDataLoaded()`

Apply this same approach for all controllers that are calling the `!safetyAlertsModel.isDataLoaded()`

Reply:
does make more sense this way with the model removed from the controllers and only in the services; updated